### DISCUSSION TITLE
How to Model Pinned Connections?

### DISCUSSION THREAD

**Discussion by ryan997997** [petrobras/ross#1109] [Q&A]
Hello, I'd like to first thank all the contributors to this open source code.

I'm trying to conduct a lateral analysis (torsional and axial degrees of freedom are excluded) for the following system:
![image](https://github.com/user-attachments/assets/dcb44fe2-7e70-4198-b5c8-9c04ba13a050)
The physical model shows a connection of a coupling to a shaft (one side only). The shaft and hub are rigidly connected. I'd like to model the flexible element as a pinned connection between the hub and spacer, i.e., it allows free rotation between the hub and spacer.

Here is my FEA model:
![image](https://github.com/user-attachments/assets/7c3a4c12-2494-49a6-8596-b869f177ed1f)
As of the current model, the hub and spacer share one common node so they are rigidly connected, so no relative rotation between the hub and spacer is allowed. To fix this, I'd like to place two nodes in that location instead of just one common node, such that these two nodes have independent rotational degrees of freedom, but no relative translational degrees of freedom (i.e. they have the same translational displacements), which should form a pinned joint.

Any ideas on how I can implement this? Is my approach correct or is there a better way to model a pinned connection? Thank you.

**Comment by jguarato** [comment]
Hi @ryan997997!

We are implementing a new element class called `CouplingElement` that may apply to your problem. It is not yet available in the latest version of ROSS, but you can check out PR #1112. We hope to include these changes in the development version of the code soon.
